### PR TITLE
Show analytics table data Date column in user's local time zone

### PR DESCRIPTION
**Pull request by snittoor**
Displays analytics table data Date column in user's local time zone.

#2183

### CODE REVIEW

File: dashboard/client/charts/charts.js [hunk 1]
```diff
@@ -263,7 +262,12 @@ Template.dashboardCharts.onCreated(function () {
       let responseStatus;
 
       // Error handling for empty fields
-      try { time = moment(e.fields.request_at[0]).toISOString(); } catch (err) { time = ''; }
+      // try { time = moment.utc(e.fields.request_at[0]).toISOString(); }
+      // catch (err) { time = ''; } original
+      let localTime = moment.utc(e.fields.request_at[0]).toISOString();
+      localTime = moment.utc(localTime).toDate();
+
+      try { time = localTime; } catch (err) { time = ''; }
```
Comment (brylie) [inline dashboard/client/charts/charts.js]: Don't use try/catch for conditional logic. Use if/else, or something similar. 

Try/catch are for error handling.

Comment (brylie) [inline dashboard/client/charts/charts.js]: Add comments describing what this code is intended to do. Why is it written?

File: users/collection/users_schema.js [hunk 2]
```diff
@@ -15,8 +15,9 @@ const UserProfileSchema = new SimpleSchema({
     optional: true,
   },
 });
-// Username must be 3-15 alphanumeric string with hyphens allowed.
-const UsernameRegEx = /^(?!\d)(?!.*-.*-)(?!.*-$)(?!-)[a-zA-Z0-9-]{3,15}$/;
+// Username must be 3-15 alphanumeric string combinations with hyphens and underscore allowed
+// Username cannot begin with a hypen , underscore
+const UsernameRegEx = /^(?!-)(?!_)[a-zA-Z0-9-_]{3,15}$/;
```
Comment (brylie) [inline users/collection/users_schema.js]: This belongs in a different pull request, and should not be part of this review.

File: dashboard/client/charts/charts.js (hunk 1 above)
Comment (snittoor) [inline dashboard/client/charts/charts.js]: The complete table is populated using try .. catch logic. So i did not alter the code. Let me know if i need to change it to if--else logic.

Comment (snittoor) [inline dashboard/client/charts/charts.js]: The complete table is populated using try .. catch logic. So i did not alter the code. Let me know if i need to change it to if--else logic.

Comment (snittoor) [inline dashboard/client/charts/charts.js]: Sorry i should have added comments.

**Comment by brylie** [comment]
This looks alright. We can ignore the `try/catch` usage here. Just be aware not to use try/catch when you need an if/else.

**Comment by brylie** [comment]
Please resolve the conflicts, and I will merge.

**Comment by nnn** [comment]
Changes look good. @snittoor please resolve the conflicts.

**Comment by snittoor** [comment]
@nnn , @brylie : Done conflict resolved , Could you please review.
